### PR TITLE
Create smwwims.md

### PR DESCRIPTION
**Pull request by smwwims**
## Describe your project in 1-3 sentences

A fully automatic cultivation machine. In contrast to the "Smart Planting Sets" available on the market, my machine is not a simple control circuit, but a regulation circuit. This means that the water and light supply is not simply controlled by the time but responds to the needs of the plants via sensors.

## Hack Club Slack handle
Simon Wimmer

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

File: smwwims.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+| Water Hose                  | https://www.amazon.de/-/en/gp/product/B07RDJY6DT/ref=ewc_pr_img_1?smid=A3JWKAKR8XB7XF&th=1   | $12.85 |
+| Plant Lamp                  | https://www.amazon.de/-/en/gp/product/B0B1MRYNWR/ref=ewc_pr_img_1?smid=A1Y73AAMUAXH83&th=1   | $25.71 |
+| Arduino Plant Watering Kit  | https://www.amazon.de/-/en/gp/product/B0BJ7W59SS/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&th=1   | $25.92 |
+| LCD Display                 | https://www.amazon.de/-/en/gp/product/B07Y4RFLZ5/ref=ewc_pr_img_1?smid=A1A7E5ILEFA1R3&psc=1  | $15.64 |
+| Breadboard PCB              | https://www.amazon.de/-/en/gp/product/B0734XYJPM/ref=ewc_pr_img_1?smid=AZF7WYXU5ZANW&psc=1   | $14.99 |
+| Main Water Pump             | https://www.amazon.de/gp/product/B071JXJ9BR/ref=ox_sc_act_title_1?smid=ADNI1MM5F189O&psc=1   | $26.78 |
+| Relay                       | https://www.amazon.de/-/en/gp/product/B07CNR7K9B/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&psc=1  | $7.49  |
+| Moisture Sensor             | https://www.amazon.de/-/en/gp/product/B07CQT5RC8/ref=ewc_pr_img_1?smid=A3BI8G9NTBZUKM&psc=1  | $6.42  |
```
Comment (bellesea) [inline smwwims.md:37]: do you intentionally want two sensors? it looks like the plant watering kit above already includes a moisture sensor?

https://www.amazon.de/-/en/gp/product/B0BJ7W59SS/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&th=1

File: smwwims.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+| Plant Pot                   | https://www.amazon.de/-/en/gp/product/B08KWPGMGY/ref=ewc_pr_img_13?smid=A2WPX7PK44TEBQ&th=1  | $41.69 |
+| Water Tank                  | https://www.amazon.de/-/en/gp/product/B0B8YSNG5C/ref=ewc_pr_img_1?smid=A3TW56H8H7ZPJ&th=1    | $26.78 |
+| Aluminium Sheet             | https://www.amazon.de/-/en/gp/product/B07XC24C9G/ref=ewc_pr_img_1?smid=A34I19W3628WJL&th=1   | $20.41 |
+| Water Hose                  | https://www.amazon.de/-/en/gp/product/B07RDJY6DT/ref=ewc_pr_img_1?smid=A3JWKAKR8XB7XF&th=1   | $12.85 |
+| Plant Lamp                  | https://www.amazon.de/-/en/gp/product/B0B1MRYNWR/ref=ewc_pr_img_1?smid=A1Y73AAMUAXH83&th=1   | $25.71 |
+| Arduino Plant Watering Kit  | https://www.amazon.de/-/en/gp/product/B0BJ7W59SS/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&th=1   | $25.92 |
+| LCD Display                 | https://www.amazon.de/-/en/gp/product/B07Y4RFLZ5/ref=ewc_pr_img_1?smid=A1A7E5ILEFA1R3&psc=1  | $15.64 |
+| Breadboard PCB              | https://www.amazon.de/-/en/gp/product/B0734XYJPM/ref=ewc_pr_img_1?smid=AZF7WYXU5ZANW&psc=1   | $14.99 |
```
Comment (bellesea) [inline smwwims.md:34]: do you need wires? also, it looks like this protoboard is for soldering? do you already have a soldering set?

File: smwwims.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Plan
+
+1. Preparing the plant pot:   Install the water tank, demarcate the electronic part
+2. Water supply:              Install pump, install water pipe
+3. Sensors:                   Wire the sensors
+4. Light:                     Install and wire the plant light
+5. Programming:               Make it all work through software
+
+## Budget
```
Comment (bellesea) [inline smwwims.md:23]: does your budget include taxes and shipping? if it doesn't, could you please include it?

File: smwwims.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+| Water Tank                  | https://www.amazon.de/-/en/gp/product/B0B8YSNG5C/ref=ewc_pr_img_1?smid=A3TW56H8H7ZPJ&th=1    | $26.78 |
+| Aluminium Sheet             | https://www.amazon.de/-/en/gp/product/B07XC24C9G/ref=ewc_pr_img_1?smid=A34I19W3628WJL&th=1   | $20.41 |
+| Water Hose                  | https://www.amazon.de/-/en/gp/product/B07RDJY6DT/ref=ewc_pr_img_1?smid=A3JWKAKR8XB7XF&th=1   | $12.85 |
+| Plant Lamp                  | https://www.amazon.de/-/en/gp/product/B0B1MRYNWR/ref=ewc_pr_img_1?smid=A1Y73AAMUAXH83&th=1   | $25.71 |
+| Arduino Plant Watering Kit  | https://www.amazon.de/-/en/gp/product/B0BJ7W59SS/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&th=1   | $25.92 |
+| LCD Display                 | https://www.amazon.de/-/en/gp/product/B07Y4RFLZ5/ref=ewc_pr_img_1?smid=A1A7E5ILEFA1R3&psc=1  | $15.64 |
+| Breadboard PCB              | https://www.amazon.de/-/en/gp/product/B0734XYJPM/ref=ewc_pr_img_1?smid=AZF7WYXU5ZANW&psc=1   | $14.99 |
+| Main Water Pump             | https://www.amazon.de/gp/product/B071JXJ9BR/ref=ox_sc_act_title_1?smid=ADNI1MM5F189O&psc=1   | $26.78 |
```
Comment (bellesea) [inline smwwims.md:35]: how do you know that this water pump is compatible with your current set up? also, it looks like the plant watering kit you included already has a pump?

**Review by bellesea — CHANGES_REQUESTED** [review]
@smwwims this is super awesome!! i'm wondering, are you following any sort of guide for this?

also, i left a few other comments in the file :)

thank you sm!

File: smwwims.md (hunk 1 above)
Comment (smwwims) [reply smwwims.md:37]: Yes, I’m planning to build a large “plant farm” (approximately 31*12 in). Therefore I need more sensors in order to get accurate results for the moisture of the soil. At least one on each side, over the soil and below the soil.

File: smwwims.md (hunk 2 above)
Comment (smwwims) [reply smwwims.md:34]: Yes, I already have a soldering set and the wires. Should I add them to the list for clarity?

File: smwwims.md (hunk 3 above)
Comment (smwwims) [reply smwwims.md:23]: Yes, the budget includes taxes and shipping. All items are from Amazon Austria, the taxes and shipping costs are already included in the prize.

File: smwwims.md (hunk 4 above)
Comment (smwwims) [reply smwwims.md:35]: The water pump is controlled via the relay. The relay is compatible with the Arduino I’m using.
I’m using two water pumps for my project: The main water pump supplies the plants with water. Below the ground is an aluminum sheet with holes. The water can run off there and flow into a catch basin. The water is pumped back from this catch basin to the water tank via the secondary pump. This protects the plants from waterlogging.

autonomous_plant_farm

**Comment by smwwims** [comment]
> @smwwims this is super awesome!! i'm wondering, are you following any sort of guide for this?
> 
> also, i left a few other comments in the file :)
> 
> thank you sm!

Thank you very much! No, I’m not following any guide. I looked into the tech specs of the smart planting farms there are available on the market for inspiration, but the overall plan is all by myself.

File: smwwims.md (hunk 2 above)
Comment (bellesea) [reply smwwims.md:34]: yes please!

**Review by bellesea — CHANGES_REQUESTED** [review]
@smwwims thank you so much for clarifying the points! could you please add what you've written into your file? perhaps elaborating under plan?

thank you :)

**Comment by smwwims** [comment]
> @smwwims thank you so much for clarifying the points! could you please add what you've written into your file? perhaps elaborating under plan?
> 
> thank you :)

@bellesea yes, of course. I updated the file with all clarifications.

**Comment by smwwims** [comment]
@bellesea this commit version contains now all the additions and clarifications we talked about and is now ready to be reviewed. Write me if I should add anything else.
Thank you very much!

**Review by bellesea — CHANGES_REQUESTED** [review]

File: smwwims.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Budget
+
+| Product                     | Supplier/Link                                                                                | Cost   |
+| --------------------------- | -------------------------------------------------------------------------------------------- | ------ |
+| Plant Pot                   | https://www.amazon.de/-/en/gp/product/B08KWPGMGY/ref=ewc_pr_img_13?smid=A2WPX7PK44TEBQ&th=1  | $41.69 |
+| Water Tank                  | https://www.amazon.de/-/en/gp/product/B0B8YSNG5C/ref=ewc_pr_img_1?smid=A3TW56H8H7ZPJ&th=1    | $26.78 |
+| Aluminium Sheet             | https://www.amazon.de/-/en/gp/product/B07XC24C9G/ref=ewc_pr_img_1?smid=A34I19W3628WJL&th=1   | $20.41 |
+| Water Hose                  | https://www.amazon.de/-/en/gp/product/B07RDJY6DT/ref=ewc_pr_img_1?smid=A3JWKAKR8XB7XF&th=1   | $12.85 |
+| Plant Lamp                  | https://www.amazon.de/-/en/gp/product/B0B1MRYNWR/ref=ewc_pr_img_1?smid=A1Y73AAMUAXH83&th=1   | $25.71 |
```
[the remark this answers is not in the document]
Comment (bellesea) [reply smwwims.md:31]: @smwwims apologies i wrote that draft and forgot to submit the review 😅

**Comment by smwwims** [comment]
@bellesea 
no, problem :). Yes, the light is also controlled via the LDR. The light is controlled with a relay, the same way as I control the water pump. The relay is compatible with the Arduino and the light is compatible with the relay.

I already updated the file with the information about the light sensors, so its ready for re-review.
Thank you.

File: smwwims.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (smwwims) [reply smwwims.md:31]: Yes, the light is also controlled via the LDR. The light is controlled with a relay, the same way as I control the water pump. The relay is compatible with the Arduino and the light is compatible with the relay.

[the remark this answers is not in the document]
Comment (bellesea) [reply smwwims.md:31]: @smwwims that sounds great! however, with the lights you linked i don't see anything about being able to programatically control the lights besides clicking on it? i might be wrong though and if so could you please link how you know this specific light would work?

[the remark this answers is not in the document]
Comment (smwwims) [reply smwwims.md:31]: @bellesea I’m going to cut through the wire and connect the wires with the relay (https://www.amazon.de/-/en/gp/product/B07CNR7K9B/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&psc=1). That way the Arduino can control the light by simply setting a digital output pin to low or high.

The Arduino is not controlling the plant light itself, but the relay. The relay works as a switch: If it gets a signal from the Arduino the circuit gets closed and the lamp can shine.

The Arduino also reads in the values from the LDR. If the brightness falls below the threshold programmed in the Arduino, the Arduino switches the plant light on. (except it’s night, then not).

**Comment by smwwims** [comment]
@bellesea I’m going to cut through the wire and connect the wires with the relay (https://www.amazon.de/-/en/gp/product/B07CNR7K9B/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&psc=1). That way the Arduino can control the light by simply setting a digital output pin to low or high.

The Arduino is not controlling the plant light itself, but the relay. The relay works as a switch: If it gets a signal from the Arduino the circuit gets closed and the lamp can shine.

The Arduino also reads in the values from the LDR. If the brightness falls below the threshold programmed in the Arduino, the Arduino switches the plant light on. (except it’s night, then not).

**Comment by bellesea** [comment]
> @bellesea I’m going to cut through the wire and connect the wires with the relay (https://www.amazon.de/-/en/gp/product/B07CNR7K9B/ref=ewc_pr_img_1?smid=A1X7QLRQH87QA3&psc=1). That way the Arduino can control the light by simply setting a digital output pin to low or high.
> 
> The Arduino is not controlling the plant light itself, but the relay. The relay works as a switch: If it gets a signal from the Arduino the circuit gets closed and the lamp can shine.
> 
> The Arduino also reads in the values from the LDR. If the brightness falls below the threshold programmed in the Arduino, the Arduino switches the plant light on. (except it’s night, then not).

ahhhh that's awesome! thank you :) could you please add this detail within your file?

**Review by bellesea — APPROVED** [review]
@smwwims this is going to be soo incredible and it looks spectacular!

an awesome member of @hackclub/bank-ops will be reaching out next week with next steps :)

(ps. please still commit the additional information but approving this to move the process forward, thank you for your patience)

thank you!

**Review by MelSmith104 — COMMENTED** [review]
hey @smwwims this sounds so neat! im a huge fan of gardening and i can't wait to hear more about your project :) im dm'ing you on slack right now to chat about next steps!

**Review by MelSmith104 — APPROVED** [review]

**Pull request merged**